// Copyright 2024 Dennis Hezel
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/* [example-proto] */
syntax = "proto3";

package fantasy.v1;

service Example {
  rpc Notice(stream NoticeRequest) returns (stream NoticeResponse) {}
  rpc GetOrderSeqNo(GetOrderSeqNoRequest) returns (GetOrderSeqNoResponse) {}
  rpc Order(OrderRequest) returns (OrderResponse) {}

  rpc ServerStreaming(OrderRequest) returns (stream OrderResponse) {}
  rpc ClientStreaming(stream OrderRequest) returns (OrderResponse) {}
}

message GetOrderSeqNoRequest {
}

message GetOrderSeqNoResponse {
  string order_seq_no = 1;
}

message OrderRequest {
  string order_seq_no = 1;
}

message OrderResponse {
  string order_seq_no = 1;
}

message NoticeRequest {
  string notice_seq_no = 1;
}

message NoticeResponse {
  string notice_seq_no = 1;
}

/* [example-proto] */
